plugins/node/opentelemetry-instrumentation-cassandra/src/instrumentation.ts

```diff
@@ -0,0 +1,362 @@
+          const originalCallback = args[
+            args.length - 1
+          ] as CassandraDriver.ValueCallback<CassandraDriver.types.ResultSet>;
+
+          const patchedCallback = function (
+            this: unknown,
+            ...cbArgs: Parameters<typeof originalCallback>
+          ) {
+            const error = cbArgs[0];
+
+            if (error) {
+              span.setStatus({
+                code: SpanStatusCode.ERROR,
+                message: error.message,
+              });
+              span.recordException(error);
+            }
+
+            span.end();
+
+            return originalCallback.apply(this, cbArgs);
+          };
+
+          args[args.length - 1] = patchedCallback;
+
+          return context.with(batchContext, () => {
+            return original.apply(this, args);
+          });
+        }
+
+        const batchPromise = safeExecuteInTheMiddle(
+          () => {
+            return original.apply(
+              this,
+              args
+            ) as Promise<CassandraDriver.types.ResultSet>;
+          },
+          error => {
+            if (error) {
+              failSpan(span, error);
+            }
+          }
+        );
+
+        const wrappedPromise = wrapPromise(span, batchPromise);
+
+        return context.bind(batchContext, wrappedPromise);
+      };
+    };
+  }
+
+  private _getPatchedStream() {
+    return (original: (...args: unknown[]) => EventEmitter) => {
+      const plugin = this;
+      return function patchedStream(
+        this: CassandraDriver.Client,
+        ...args: unknown[]
+      ) {
+        // Since stream internally uses execute, there is no need to add DB_STATEMENT twice
+        const span = plugin.startSpan({ op: 'stream' }, this);
+
+        const callback = args[3];
+
+        const endSpan = (error: Error) => {
+          if (error) {
+            span.setStatus({
+              code: SpanStatusCode.ERROR,
+              message: error.message,
+            });
+            span.recordException(error);
+          }
+          span.end();
+        };
+
+        if (callback === undefined) {
+          args[3] = endSpan;
+        } else if (typeof callback === 'function') {
+          const wrappedCallback = function (this: unknown, err: Error) {
+            endSpan(err);
+            return callback.call(this, err);
+          };
+          args[3] = wrappedCallback;
+        }
+
+        return safeExecuteInTheMiddle(
+          () => {
+            return original.apply(this, args);
+          },
+          error => {
+            if (error) {
+              failSpan(span, error);
+            }
+          }
+        );
+      };
+    };
+  }
+
+  public startSpan(
+    { op, query }: { op: string; query?: unknown },
+    client: CassandraDriver.Client
+  ): Span {
+    const attributes: SpanAttributes = {
+      [SemanticAttributes.DB_SYSTEM]: DbSystemValues.CASSANDRA,
+    };
+
+    if (this._shouldIncludeDbStatement() && query !== undefined) {
```

Comment:
Standard PII concerns here -
- What normalization/sanitization is/should be performed on this string/set of strings?  Might it contain sensitive values?
- If it can contain sensitive values, then the documentation about the config option should note the risk.

Reply:
It can contain sensitive information if the queries are are not escaped, e.g. when using
```javascript
const creditCard = '1234';
client.execute(`insert into credit_cards (number) values ('${creditCard}')`);
```
instead of the usual interface
```javascript
client.execute('insert into credit_cards (number) values (?)', ['1234']);
```

With the first case you'd have more to worry about anyway :smile:

Anyway I added a comment to both code and README, thanks!

Reply:
Since enhanced reporting is off by default and the behavior is documented I think this is acceptable thanks.

Reply:
Seconded Daniel's viewpoint.  LGTM now